src/arch/riscv/common_riscv.lds

```diff
@@ -43,12 +43,13 @@ SECTIONS
 
     /* Start of data section */
     _sdata = .;
-    .sdata : {
+    .small : {
+        /* Small data that should be accessed relative to gp.  ld has trouble
+           with the relaxation if they are not in a single section. */
         __global_pointer$ = . + 0x800;
-        *(.sdata*)
-    }
-    .srodata : {
         *(.srodata*)
```

Comment:
Is it intended that `.srodata {...}` is dropped and `*(.srodata*)` is part of `.small` now?